goship.go

```diff
@@ -321,6 +331,7 @@ func DeployOutputHandler(w http.ResponseWriter, r *http.Request, env string, for
 }
 
 func DeployLogHandler(w http.ResponseWriter, r *http.Request, full_env string, environment goship.Environment) {
+	u, _ := getUser(r)
```

Comment:
I log the error in the getUser func itself.  checkAuth actually needs the err so I return it, other functions that use it do not need to know an error occurs. They actually get a "fake user" when auth is turned off. Do you have a better suggestion on how to handle it?